darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,235 @@
+import os
+import argparse
+import datetime
+import importlib.resources as importlib_resources
+from typing import Any, Union
+
+import pandas as pd
+from mako.template import Template
+
+import darshan
+import darshan.cli
+
+
+class ReportData:
+    """
+    Collects all of the metadata, tables, and figures
+    required to generate a Darshan Summary Report.
+
+    Parameters
+    ----------
+    log_path: path to a darshan log file.
+
+    """
+    def __init__(self, log_path: str):
+        # store the log path and use it to generate the report
+        self.log_path = log_path
+        # store the report
+        self.report = darshan.DarshanReport(log_path, read_all=True, dtype="pandas")
```

Comment:
I think this call signature with dtype specification is the exact one that @jakobluettgau wanted us to avoid per discussion this afternoon and gh-490, so probably good to open an issue for this specific usage.

Reply:
ok, I opened the issue...